phpunit/experimental/interactivity-api/directive-processing-test.php

```diff
@@ -73,73 +73,43 @@ public function test_directives_with_double_hyphen_processed_correctly() {
-		$tags   = new WP_HTML_Tag_Processor( $markup );
-		gutenberg_interactivity_process_rendered_html( $tags, 'foo-', $directives );
+		$tags   = new WP_Directive_Processor( $markup );
+		$tags->process_rendered_html( $tags, 'foo-', $directives );
 	}
 
 	public function test_interactivity_process_directives_in_root_blocks() {
-		$pattern_content =
+
+		$block_content =
 		'<!-- wp:paragraph -->' .
-			'<p>Pattern Content Block 1</p>' .
+			'<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>' .
 		'<!-- /wp:paragraph -->' .
 		'<!-- wp:paragraph -->' .
-			'<p>Pattern Content Block 2</p>' .
+			'<p>Welcome to WordPress.</p>' .
 		'<!-- /wp:paragraph -->';
-		register_block_pattern(
-			'core/interactivity-pattern',
-			array(
-				'title'   => 'Interactivity Pattern',
-				'content' => $pattern_content,
-			)
-		);
 
-		$providers = $this->data_only_root_blocks_are_processed();
-		foreach ( $providers as $provider ) {
-			do_blocks( $provider['page_content'] );
-			$this->assertSame( $provider['root_blocks'], count( WP_Directive_Processor::$root_blocks ) );
+		$parsed_block     = parse_blocks( $block_content )[0];
+		$rendered_content = render_block( $parsed_block );
 
-		}
-	}
+		$parsed_block_second = parse_blocks( $block_content )[1];
 
-	/**
-	 * Data provider .
-	 *
-	 * @return array
-	 **/
-	public function data_only_root_blocks_are_processed() {
+		// Test that root block is intially emtpy.
+		$this->assertEmpty( WP_Directive_Processor::$root_block );
 
-		return array(
-			array(
-				'root_blocks'  => 2,
-				'page_content' =>
-				'<!-- wp:quote -->' .
-					'<blockquote class="wp-block-quote">
-						<!-- wp:paragraph -->' .
-						'<p>The XYZ Doohickey Company was founded in 1971, and has been providing' .
-						'quality doohickeys to the public ever since. Located in Gotham City, XYZ employs' .
-						'over 2,000 people and does all kinds of awesome things for the Gotham community.</p>' .
-						'<!-- /wp:paragraph -->
-					</blockquote>' .
-				'<!-- /wp:quote -->' .
-				'<!-- wp:quote -->' .
-					'<blockquote class="wp-block-quote">
-						<!-- wp:paragraph -->' .
-						'<p>The XYZ Doohickey Company was founded in 1971, and has been providing' .
-						'quality doohickeys to the public ever since. Located in Gotham City, XYZ employs' .
-						'over 2,000 people and does all kinds of awesome things for the Gotham community.</p>' .
-						'<!-- /wp:paragraph -->
-					</blockquote>' .
-				'<!-- /wp:quote -->',
-			),
-			array(
-				'root_blocks'  => 2,
-				'page_content' =>
-				'<!-- wp:paragraph -->' .
-					'<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>' .
-				'<!-- /wp:paragraph -->' .
-				'<!-- wp:pattern {"slug":"core/interactivity-pattern"} /-->',
-			),
-		);
+		// Test that root block is not added if there is a parent block.
+		gutenberg_interactivity_process_directives( $parsed_block, $parsed_block, $parsed_block );
```

Comment:
Passing `$parsed_block` three times is confusing.

```suggestion
		$fake_parent_block = array();
		$source_block = $parsed_block;
		gutenberg_interactivity_process_directives( $parsed_block, $source_block, $fake_parent_block );
```